src/main/java/org/cftoolsuite/cfapp/notifier/EmailNotifier.java

```diff
@@ -75,17 +75,17 @@ public void onApplicationEvent(EmailNotificationEvent event) {
         String footer =
             String.format("This email was sent from %s on %s",
                     event.getDomain(), DateTimeFormatter.ISO_DATE_TIME.format(LocalDateTime.now()));
-        String subject = event.getSubject();
+        String subject = String.format("%s , Region: %s", event.getSubject(), event.getDomain());
         final String body = buildBody(template, event.getBody(), subject, footer);
         log.trace("About to send email using ||> From: {}, To: {}, Cc: {}, Bcc: {}, Subject: {}, Body: {}", from, recipients.toString(), String.join(", ", cc), String.join(", ", bcc), subject, body);
         List<EmailAttachment> prunedAttachments = new ArrayList<EmailAttachment>();;
         if (event.getAttachments() != null){
             prunedAttachments = event.getAttachments().stream().filter(EmailAttachment::hasContent).collect(Collectors.toList());
         }
         List<EmailAttachment> attachments = prunedAttachments;
-        recipients.forEach(recipient -> {
-            sendMail(from, recipient, cc, bcc, subject, body, attachments);
-        });
+
+        String[] toRecipients = recipients.toArray(new String[0]);
+        sendMail(from, toRecipients, cc, bcc, subject, body, attachments);
```

Comment:
This change you've suggested, necessitates some further cascading changes to the `abstract sendMail` method signature and then the implementers: `JavaMailNotitifier` and `SendMailNotifier`.  It's trivial to change the signature of the `sendMail` method in `JavaMailNotifier` because `MimeMessageHelper`'s `setTo` method also accepts `String[]`.  However, updating `SendMailNotifier`'s `sendMail` method logic required some adjustment to the way you pass multiple recipients defined for to.

These are the relevant lines I had to update in `SendMailNotifier`'s `sendMail` method:

```
@Override
    public void sendMail(String originator, String recipients[], String[] carbonCopyRecipients, String[] blindCarbonCopyRecipients, String subject, String body, List<EmailAttachment> attachments) {
        try {
            Email from = new Email(originator);
            Content content = new Content("text/html", body);
            Mail mail = new Mail();
            Personalization personalization = new Personalization();
            personalization.setFrom(from);
            if (recipients != null && recipients.length > 0) { Arrays.asList(recipients).forEach(to -> personalization.addCc(new Email(to))); }
```

Reply:
Please include the relevant changes in your next commit for this PR, and I'll have another look.

We also don't have test coverage for the implementers. We (I) really should have drafted some tests for variant cases, like:

No subject
A subject
No recipients (to)
One or multiple recipients (to)
No from
One from
None, one or multiple cc
None, one or multiple bcc
None, one or multiple attachments

Here's something to get us started: 

https://www.perplexity.ai/search/i-have-the-following-abstract-aNv0JDeJQq.YjnAH0AME6Q

Reply:
Thank you for taking time to explain these. I will try to add them to the next MR.

Reply:
I've made new commits according to your suggestions above!